### PR TITLE
[RW-11933][risk=low] API support autodelete

### PR DESCRIPTION
**Pull request by yonghaoy**
---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/resources/leonardo.yaml [hunk 1]
```diff
@@ -2489,6 +2489,13 @@ components:
           items:
             type: string
           description: Extra arguments to pass to the application. Only used if appType is CUSTOM.
+        autodeleteThreshold:
+          type: integer
+          description: The number of minutes of idle time to elapse before the app is
+            deleted. When autodeleteEnabled is true, a positive integer is required
+        autodeleteEnabled:
```
Comment (yonghaoy) [inline api/src/main/resources/leonardo.yaml:2496]: this is copied from Leo

File: api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java [hunk 2]
```diff
@@ -604,7 +604,11 @@ public void createApp(CreateAppRequest createAppRequest, DbWorkspace dbWorkspace
         .diskConfig(diskRequest)
         .customEnvironmentVariables(appCustomEnvVars)
         .workspaceId(dbWorkspace.getFirecloudUuid())
-        .labels(appLabels);
+        .labels(appLabels)
+        .autodeleteThreshold(createAppRequest.getAutodeleteThreshold());
+    if(createAppRequest.isAutodeleteEnabled()) {
```
Comment (yonghaoy) [inline api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java:609]: same to autopause below

**Review by tarekmamdouh — APPROVED** [review]
Looks good, maybe rerun the e2e tests, I believe the error there is irrelevant

File: api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java [hunk 3]
```diff
@@ -604,7 +604,11 @@ public void createApp(CreateAppRequest createAppRequest, DbWorkspace dbWorkspace
         .diskConfig(diskRequest)
         .customEnvironmentVariables(appCustomEnvVars)
         .workspaceId(dbWorkspace.getFirecloudUuid())
-        .labels(appLabels);
+        .labels(appLabels)
+        .autodeleteThreshold(createAppRequest.getAutodeleteThreshold());
+    if (createAppRequest.isAutodeleteEnabled() != null) {
```
[the remark this answers is not in the document]
Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java:609]: Good catch: https://github.com/all-of-us/workbench/pull/8379

**Pull request merged**